targets/arm/mikroe/sam/src/adc/implementations/implementation_1/hal_ll_adc.c

```diff
@@ -0,0 +1,546 @@
+
+        low_level_handle->hal_ll_adc_handle = NULL;
+        low_level_handle->hal_drv_adc_handle = NULL;
+
+        low_level_handle->init_ll_state = false;
+
+        hal_ll_adc_hw_specifics_map_local->pin = HAL_LL_PIN_NC;
+    }
+}
+
+// ----------------------------------------------- PRIVATE FUNCTION DEFINITIONS
+static hal_ll_pin_name_t hal_ll_adc_check_pins( hal_ll_pin_name_t pin, hal_ll_adc_pin_id *index, hal_ll_adc_handle_register_t *handle_map )
+{
+    static const uint16_t     adc_map_size = ( sizeof( _adc_map ) / sizeof( hal_ll_adc_pin_map_t ) );
+    uint16_t     pin_index = 0;
+    uint8_t      index_counter = 0;
+    uint8_t      hal_ll_module_id = 0;
+
+    if ( HAL_LL_PIN_NC == pin ) {
+        return HAL_LL_PIN_NC;
+    }
+
+    for ( pin_index = 0; pin_index < adc_map_size; pin_index++ )
+    {
+        if ( pin == _adc_map[pin_index].pin )
+        {
+            // Get module number
+            hal_ll_module_id = _adc_map[pin_index].module_index;
+            // Map pin name
+            index->pin_an[hal_ll_module_id] = pin_index;
+
+            // Check if module is taken
+            if ( NULL == handle_map[ hal_ll_module_id ].hal_drv_adc_handle ) {
+                return hal_ll_module_id;
+            } else if ( ADC_MODULE_COUNT == ++index_counter ) {
+                return --index_counter;
+            }
+        }
+    }
+
+    if ( index_counter ) {
+        return hal_ll_module_id;
+    } else {
+        return HAL_LL_PIN_NC;
+    }
+}
+
+static void hal_ll_adc_map_pin( uint8_t module_index, hal_ll_adc_pin_id *index ) {
+    // Map new pins
+    hal_ll_adc_hw_specifics_map[module_index].pin = _adc_map[ index->pin_an[module_index] ].pin;
+    hal_ll_adc_hw_specifics_map[module_index].channel = _adc_map[ index->pin_an[module_index] ].channel;
+}
+
+static hal_ll_adc_hw_specifics_map_t *hal_ll_get_specifics( handle_t handle ) {
+    uint8_t hal_ll_module_count = sizeof( hal_ll_module_state ) / ( sizeof( hal_ll_adc_handle_register_t ) );
+    static uint8_t hal_ll_module_error = sizeof( hal_ll_module_state ) / ( sizeof( hal_ll_adc_handle_register_t ) );
+
+    while( hal_ll_module_count-- ) {
+        if ( hal_ll_adc_get_base_from_hal_handle == hal_ll_adc_hw_specifics_map[hal_ll_module_count].base ) {
+            return &hal_ll_adc_hw_specifics_map[hal_ll_module_count];
+        }
+    }
+
+    return &hal_ll_adc_hw_specifics_map[hal_ll_module_error];
+}
+
+static inline void adc0_enable_clock()
+{
+    set_reg_bit( _PMC_PCER0, HAL_LL_PID_AFEC0_BIT );
+}
+
+static inline void adc1_enable_clock()
+{
+    set_reg_bit( _PMC_PCER1, HAL_LL_PID_AFEC1_BIT );
+}
+
+static void _hal_ll_adc_enable_clock( uint8_t module_index )
+{
+    switch ( module_index )
+    {
+        #ifdef ADC_MODULE_0
+        case ( hal_ll_adc_module_num( ADC_MODULE_0 ) ):
+            adc0_enable_clock();
+            break;
+        #endif
+        #ifdef ADC_MODULE_1
+        case ( hal_ll_adc_module_num( ADC_MODULE_1 ) ):
```

Comment:
Coding standard must ensure a default case in every switch. Add it.

Reply:
Added it for all source files where needed.